Resendcount

idstam opened the pull request:
Since we can not be sure all files have arrived and to enable connecting to a live stream I've added two flags
resendcount - How many times to resend all files
resendmanifest - Resend the manifest in between every file

idstam committed:
Add a flag to enable X amounts of re-transmissions.

idstam committed:
Do not delete tmpDir on manifest received

idstam committed:
Resend manifest after every file

klockcykel commented:
@idstam Nice! To offload unnecessary processing may I suggest adding a check if the manifest already been received in full by checking the manifestId against the last fully received manifest early in Receiver.onManifestPacket?

What do you mean with "connecting to a live stream"? That the receiver is (re-)started mid transfer?

I wasn't aware anyone was using this code. I know of someone that has made additional improvements and bug fixes a while back, I'll see if he can create a pull request for those fixes as well.

idstam committed:
Handle acknowledge lastManifestId when receiving manifestPackage

henryreed commented:
Looks like `resendcount` must be set to at least 1, otherwise no files are sent via the default command line in the README:

```bash
./bin/godiode --verbose --baddr 10.72.0.1:1234 send out/
```

This works though

```bash
./bin/godiode --verbose --resendcount 1 --baddr 10.72.0.1:1234 send out/
```

We may want to add a default value of 1 if a `resendcount` is not given.

idstam commented:
I have a fork that is 41 commits ahead of this where I've added a lot of things needed for production use. There I fixed this issue in config.go by setting it to one when initializing the config struct.
